mypy/fastparse.py

```diff
@@ -521,40 +521,45 @@ def translate_stmt_list(
             return [block]
 
         stack = self.class_and_function_stack
-        if self.strip_function_bodies and len(stack) == 1 and stack[0] == "F":
+        # Fast case for stripping function bodies
+        if (
+            can_strip
+            and self.strip_function_bodies
+            and len(stack) == 1
+            and stack[0] == "F"
+            and not is_coroutine
```

Comment:
We can strip if there is no `yield` right? Also IIUC this is the actual fix, while the change below is just a refactoring, right?

Reply:
Change below is actual fix too. The stripping below happened only in methods

Reply:
OK, wait, but then now the stripping in last part will occur even if `stack[0] != "F"`, is it safe?

Reply:
Added that check, and fixed a bug I introduced in the logic for empty bodies